### ISSUE TITLE
Manta M8P + CM4104032 + BTT PITFT50 V2.0

### ISSUE THREAD

**Issue by Gimran** [bigtreetech/Manta-M8P#41]
Hi.
I bought the kit first with CB1, did not notice that the module does not support DSI. 
I later purchased the CM4104032 specifically to support the screen.
But when I connect BTT PITFT50 V2.0 - RPi CM4 does not start- the power LED lights up and goes out. and I hear faint clicks.
Maybe there is not enough power 3,3V for the display, or it is a BAG?
The screen on the RaspberryPI 4 works, the CM4 without the screen works, the camera also works without problems on the Manta m8p.

**Comment by MikeEitel** [comment]
Yes, that is as it is in original state, and I also quite some days...
You have to adapt the overlays AND recompile therefore.

BIGTREETECH:
sudo wget https://datasheets.raspberrypi.com/cmio/dt-blob-disp1-cam1.bin -O /boot/dt-blob.bin
​dt-blob.bin                                    must be in /boot
dt-blob-disp1-cam1.bin                must be in /boot/overlays

BUT BIGTREETECH FORGET TO WRITE:

"
RPI:
In order to compile a Device Tree source (.dts) file into a Device Tree blob (.dtb) file, the Device Tree compiler must be installed by running sudo apt install device-tree-compiler. The dtc command can then be used as follows:
"

sudo dtc -I dts -O dtb -o /boot/dt-blob.bin dt-blob.dts /boot/overlays/dt-blob-disp1-cam1.dts

Only after that I was able to have DSI1 and DCI1.
Libcam sees the camera but streamer is not jet working in klipper.... ( work in progress as I understand )

Good luck
Mike

**Comment by igetio** [comment]
I had this issue and it was caused by a faulty cable between the pitft and the m8p

**Comment by smplman** [comment]
I have a CM4 in my M8P and I had to get the blob file as outlined in the docs. As soon as I did that my display started working.
```
sudo wget https://datasheets.raspberrypi.com/cmio/dt-blob-disp1-cam1.bin -O /boot/dt-blob.bin
````
https://www.raspberrypi.com/documentation/computers/compute-module.html#quickstart-guide-display-and-cameras

**Comment by admilsonmarques** [comment]
I still have this issue... in my case the display is never recognized, although the same cable and screen is instantly recognized on pi3b+

**Comment by Ara-Nai** [comment]
I can confirm that the command  @admilsonmarques posted works: 

sudo wget https://datasheets.raspberrypi.com/cmio/dt-blob-disp1-cam1.bin -O /boot/firmware/dt-blob.bin

( with Debian 12 / bookworm. Installed Pi OS lite, Klipper/Mainsail with Kiauh on Manta M8P 2.0 / CM4 / Pi TFT50 2.1 ) 

But dont change the boot/firmware/config.txt ! Klipperscreen wont work

**Comment by Tuxedo-VE** [comment]
> I had this issue and it was caused by a faulty cable between the pitft and the m8p

Correct. I had the same problem. This is caused by an error in the schematic: the pin assignments for the 15-pin DSI header on the Manta M8P V2.0 are inverted when compared to the normal pin assignments on a Raspberry Pi's CSI header. Because of this, you cannot use the 15 conductor Type 2 FFC flat cable that comes with the Pi CAM module (exposed contacts are on opposite sides of the flat cable). You must use a Type 1 FFC cable with exposed contacts on the same side of the cable.

If you use a Type 2 FFC, the VCC and GND pins are reversed and that shorts the supply. The printer won't even boot and the camera module will get boiling hot and eventually fry. BTT needs to fix this with a new board release or put a specific warning in the Manta M8P V2.0 user manual to this effect.

**Comment by hazama6** [comment]
> > I had this issue and it was caused by a faulty cable between the pitft and the m8p
> 
> Correct. I had the same problem. This is caused by an error in the schematic: the pin assignments for the 15-pin DSI header on the Manta M8P V2.0 are inverted when compared to the normal pin assignments on a Raspberry Pi's CSI header. Because of this, you cannot use the 15 conductor Type 2 FFC flat cable that comes with the Pi CAM module (exposed contacts are on opposite sides of the flat cable). You must use a Type 1 FFC cable with exposed contacts on the same side of the cable.
> 
> If you use a Type 2 FFC, the VCC and GND pins are reversed and that shorts the supply. The printer won't even boot and the camera module will get boiling hot and eventually fry. BTT needs to fix this with a new board release or put a specific warning in the Manta M8P V2.0 user manual to this effect.

The connector on V2.0 is reversible at least it is on mine

**Comment by Tuxedo-VE** [comment]
> The connector on V2.0 is reversible at least it is on mine

Thanks for the tip. I didn't know that and BTT makes absolutely no mention of it in their manual. I like their controllers but frankly the documentation could use some improvement.

**Comment by voluble** [comment]
Apologies for bringing up an old thread. I am trying to get at TFT50 screen to work with my Manta M8p v2 and CM4. 

Does anyone have another link to the dt-blob-disp1-cam1.bin file? It seems that RaspberryPi has removed it from their website. Thanks
